### ISSUE TITLE
Unit tests for gRPC endpoints and CLI commands

### ISSUE THREAD

**Issue by ryanchristo** [regen-network/regen-ledger#1041]
## Summary

We are currently using the integration test suite for gRPC endpoints and CLI commands. This has proven difficult to maintain and introduces nuances when checking the length of returned items from queries. Integration tests might not be the best route for gRPC and CLI commands but rather used only with the server implementation methods (or maybe this too will become redundant with the implementation of acceptance tests).

## Problem Definition

How to write integration tests is not clear and the client integration tests are difficult to maintain and do not allow us to properly check the length of returned values.

## Proposal

Update the client integration tests to unit tests, removing the use of the integration test suite.

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by aaronc** [comment]
I agree integration tests should mostly be focused on the server level using `MsgClient`/`MsgServer` directly - basically the level just above unit tests.

I guess we could unit test CLI/gRPC with mocks?

I would note that own of my goals in the Cosmos SDK this year is to deprecate manually written CLI and gRPC query methods. Ideally all of that would be auto-generated by reflection on `Msg`s and ORM definitions in the .proto fiels.

**Comment by ryanchristo** [comment]
Also relevant to https://github.com/cosmos/cosmos-sdk/issues/12696
